File: backend-spring/src/main/java/com/codequest/util/LanguageDetect.java
package com.codequest.util;

public final class LanguageDetect {
    private LanguageDetect() {}

    public static boolean isPrimarilyRussian(String text) {
        if (text == null || text.isBlank()) {
            return false;
        }
        int cyrillic = 0;
        int latin = 0;
        for (char ch : text.toCharArray()) {
            if (ch >= '\u0400' && ch <= '\u04FF') {
                cyrillic++;
            } else if (Character.isLetter(ch) && ch < 128) {
                latin++;
            }
        }
        return cyrillic > 0 && cyrillic >= latin;
    }
}

File: CodeQuestApp/app/src/main/java/com/prolearn/codecraftfront/util/LanguageDetect.kt
package com.prolearn.codecraftfront.util

/**
 * Detect whether the user message is primarily Russian (Cyrillic) or English/Latin.
 */
object LanguageDetect {
    fun isPrimarilyRussian(text: String): Boolean {
        var cyrillic = 0
        var latin = 0
        for (ch in text) {
            when {
                ch in '\u0400'..'\u04FF' -> cyrillic++
                ch.isLetter() && ch.code < 128 -> latin++
            }
        }
        return cyrillic > 0 && cyrillic >= latin
    }

    fun replyLanguageInstruction(userMessage: String): String {
        return if (isPrimarilyRussian(userMessage)) {
            "Reply in Russian."
        } else {
            "Reply in English."
        }
    }
}
